Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -600,62 +307,202 @@ namespace gdjs {
             break;
           }
           object.setY(
-            Math.floor(object.getY()) + (requestedDeltaY > 0 ? -1 : 1)
+            Math.floor(object.getY()) + (this._requestedDeltaY > 0 ? -1 : 1)
           );
         }
       }
+    }
 
-      //3) Update the current floor data for the next tick:
-      this._updateOverlappedJumpThru();
-      if (!this._isOnLadder) {
-        //Check if the object is on a floor:
-        //In priority, check if the last floor platform is still the floor.
+    checkTransitionOnLadder() {
+      if (this._ladderKey && this._isOverlappingLadder()) {
+        this._setOnLadder();
+      }
+    }
+
+    checkTransitionJumping() {
+      if (this._canJump && this._jumpKey) {
+        this._setJumping();
+      }
+    }
+
+    checkGrabPlatform() {
+      const object = this.owner;
+      let tryGrabbingPlatform = false;
+      object.setX(
+        object.getX() +
+          (this._requestedDeltaX > 0
+            ? this._xGrabTolerance
+            : -this._xGrabTolerance)
+      );
+      let collidingPlatform = this._getCollidingPlatform();
+      if (collidingPlatform !== null && this._canGrab(collidingPlatform)) {
+        tryGrabbingPlatform = true;
+      }
+      object.setX(
+        object.getX() +
+          (this._requestedDeltaX > 0
+            ? -this._xGrabTolerance
+            : this._xGrabTolerance)
+      );
+
+      //Check if we can grab the collided platform
+      if (tryGrabbingPlatform) {
         let oldY = object.getY();
-        object.setY(object.getY() + 1);
+        object.setY(
+          // @ts-ignore - collidingPlatform is guaranteed to be not null.
+          collidingPlatform.owner.getY() +
+            // @ts-ignore - collidingPlatform is guaranteed to be not null.
+            collidingPlatform.getYGrabOffset() -
+            this._yGrabOffset
+        );
         if (
-          this._isOnFloor &&
-          gdjs.RuntimeObject.collisionTest(
-            object,
-            this._floorPlatform.owner,
-            this._ignoreTouchingEdges
+          !this._isCollidingWith(
+            this._potentialCollidingObjects,
+            null,
+            /*excludeJumpthrus=*/
+            true
           )
         ) {
-          //Still on the same floor
-          this._floorLastX = this._floorPlatform.owner.getX();
-          this._floorLastY = this._floorPlatform.owner.getY();
+          this._setGrabbingPlatform(collidingPlatform!);
+          this._requestedDeltaY = 0;
+        } else {
+          object.setY(oldY);
+        }
+      }
+    }
+
+    checkTransitionOnFloorOrFalling() {
+      const object = this.owner;
+      //Check if the object is on a floor:
+      //In priority, check if the last floor platform is still the floor.
+      let oldY = object.getY();
+      object.setY(object.getY() + 1);
+      if (
+        this._state == this._onFloor &&
```

Comment:
Always use === in JS. == is doing type coercion and is slower and less safe. === is always the way to go :) 
(check this for all the `this._state ==` in the file)